From b160c4add112ec90a811726ee3ac536f98cfd6cf Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Tue, 1 Nov 2011 22:02:36 +0100
Subject: [PATCH] cgroup: always recreate cgroup before we try to apply
 attributes

We might have trimmed the cgroup tree previously, hence don't trust our
own "realized" flag, always recreate cgroup tree before applying our
attributes to make sure this actually works out.

https://bugzilla.redhat.com/show_bug.cgi?id=749687
(cherry picked from commit 6ddaf1ca4ab0e6a094f6d37fa1e0c604c6d867ba)
---
 src/cgroup.c |    3 ---
 1 files changed, 0 insertions(+), 3 deletions(-)

diff --git a/src/cgroup.c b/src/cgroup.c
index dcf2c2f..be837c3 100644
--- a/src/cgroup.c
+++ b/src/cgroup.c
@@ -38,9 +38,6 @@ int cgroup_bonding_realize(CGroupBonding *b) {
         assert(b->path);
         assert(b->controller);
 
-        if (b->realized)
-                return 0;
-
         r = cg_create(b->controller, b->path);
         if (r < 0) {
                 log_warning("Failed to create cgroup %s:%s: %s", b->controller, b->path, strerror(-r));
-- 
1.7.7.5

